src/main/java/software/amazon/cloudwatchlogs/emf/model/MetricsContext.java

```diff
@@ -203,36 +203,72 @@ public MetricsContext createCopyWithContext() {
      * metrics in one log event. If there're more than 100 metrics, we split the metrics into
      * multiple log events.
      *
+     * <p>If a metric has more than 100 data points, we also split the metric.
+     *
      * @return the serialized strings.
      * @throws JsonProcessingException if there's any object that cannot be serialized
      */
     public List<String> serialize() throws JsonProcessingException {
-        if (rootNode.metrics().size() <= Constants.MAX_METRICS_PER_EVENT) {
+        if (rootNode.metrics().size() <= Constants.MAX_METRICS_PER_EVENT
+                && !anyMetricWithTooManyDataPoints(rootNode)) {
             return Arrays.asList(this.rootNode.serialize());
         } else {
             List<RootNode> nodes = new ArrayList<>();
             Map<String, MetricDefinition> metrics = new HashMap<>();
-            int count = 0;
-            for (MetricDefinition metric : rootNode.metrics().values()) {
-                metrics.put(metric.getName(), metric);
-                count++;
+            Queue<MetricDefinition> metricDefinitions =
+                    new LinkedList<>(rootNode.metrics().values());
+            while (metricDefinitions.size() > 0) {
+                MetricDefinition metric = metricDefinitions.peek();
```

Comment:
Do we need `peek()` if we're going to `poll()` regardless? Can we just `poll()` here?